src/main/scala/org/broadinstitute/dsde/vault/services/generic/GenericDescribeService.scala

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.dsde.vault.services.generic
+
+import com.wordnik.swagger.annotations._
+import org.broadinstitute.dsde.vault.common.directives.VersioningDirectives._
+import org.broadinstitute.dsde.vault.model.GenericJsonProtocol._
+import org.broadinstitute.dsde.vault.model.{GenericRelationship, GenericRelEnt, GenericEntity, GenericSysAttrs}
+import spray.http.MediaTypes._
+import spray.json._
+import spray.routing.HttpService
+
+@Api(value="/entities", description="generic entity service", produces="application/json")
+trait GenericDescribeService extends HttpService {
+  private final val ApiPrefix = "entities"
+  private final val ApiVersions = "v1"
+  private final val DefaultVersion = 1
+
+  // up and down are more specific matches of the same route, so they must be listed first
+  val gdRoutes = findUpstreamRoute ~ findDownstreamRoute ~ describeRoute
+
+  @ApiOperation(
+    value="get data for a particular entity",
+    nickname="fetchEntity",
+    httpMethod="GET",
+    response=classOf[GenericEntity])
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name="version", required=true, dataType="string", paramType="path", value="API version", allowableValues=ApiVersions),
+    new ApiImplicitParam(name="id", required=true, dataType="string", paramType="path", value="vault ID")))
+  @ApiResponses(Array(
+    new ApiResponse(code=200, message="Successful"),
+    new ApiResponse(code=404, message="Not Found"),
+    new ApiResponse(code=500, message="Internal Error")))
+  def describeRoute = {
+    pathVersion(ApiPrefix,DefaultVersion,Segment) { (version, guid) =>
+      get {
+        respondWithMediaType(`application/json`) {
+          complete {
+            //STUB
+
+            // interact with BOSS, receive getUrl
+            // remove entity.sysAttrs.bossId, add entity.getUrl
+
+            val sysAttrs = GenericSysAttrs(bossID = None, 12345, "stub user", None, None)
+            GenericEntity("entity 3 guid", None, Option("entity 3 getUrl"), "stub", sysAttrs, None).toJson.prettyPrint
+          }
+        }
+      }
+    }
+  }
+
+  @ApiOperation(
+    value="get entities upstream of a specified entity",
+    nickname="findUpstream",
+    httpMethod="GET",
+    response=classOf[GenericRelEnt],
+    responseContainer="List")
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name="version", required=true, dataType="string", paramType="path", value="API version", allowableValues=ApiVersions),
+    new ApiImplicitParam(name="id", required=true, dataType="string", paramType="path", value="vault ID"),
+    new ApiImplicitParam(name="up", required=true, dataType="string", defaultValue="up", paramType="query", value="marker for upstream query")))
```

Comment:
see DSDEV-2017 ... is there an (easy) way to get this to show as the "headline" for the api in swagger, e.g. "GET /entities/{version}/{id}?up"?  If this isn't easy, definitely defer and we'll tackle it later in DSDEV-2017.